src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private T[] array;
+    private int arraySize = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
+    private void resize() {
+        int newSize = array.length + (array.length / 2);
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (arraySize == array.length) {
+            resize();
+        }
+        array[arraySize] = value;
+        arraySize++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > arraySize) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
```

Comment:
Make your exceptions informative. Add more details to the exception message, like the invalid index and the valid range.